hw/ip/clkmgr/data/clkmgr.sv.tpl

```diff
@@ -285,6 +290,42 @@
 
   assign pwr_o.clk_status = clk_status;
 
+  ////////////////////////////////////////////////////
+  // Clock Measurement for the roots
+  ////////////////////////////////////////////////////
+
+<% aon_freq = clocks.all_srcs['aon'].freq %>\
+% for src in typed_clocks.rg_srcs:
+  logic ${src}_meas_valid;
+  logic ${src}_fast_err;
+  logic ${src}_slow_err;
+  <%
+   freq = clocks.all_srcs[src].freq
+   cnt = int(freq*2 / aon_freq)
```

Comment:
I assume the times 2 is to create a wider counter to detect much faster clock ratios.

Reply:
yes that's right. Let me clarify that somewhere.